internal/shared/http.go

```
@@ -328,6 +338,37 @@ func FindModelInPath(cfg config.Config, path string) (searchName, realName, rema
 	return
 }
 
+// EscapedPathSuffix removes decodedPrefix from escapedPath while leaving the
+// remaining percent-encoding untouched. Decoded paths cannot identify the
+// boundary alone: a model name such as "author/model" may have arrived as the
+// single escaped segment "author%2Fmodel".
+func EscapedPathSuffix(escapedPath, decodedPrefix string) string {
+	rawIndex, prefixIndex := 0, 0
+	for rawIndex < len(escapedPath) && prefixIndex < len(decodedPrefix) {
+		end := rawIndex + 1
```

golangci-lint (2.12.2)

[error] ineffectual assignment to end

(ineffassign)